src/com/gfredericks/test/chuck/generators.clj

```diff
@@ -252,3 +254,54 @@
   (gen/fmap (fn [ks]
               (select-keys m ks))
             (subsequence (keys m))))
+
+(def ^:private valid-offset-fns [ct/millis ct/seconds ct/minutes ct/hours ct/days ct/months ct/years])
+(def ^:private valid-offset-fns? (set valid-offset-fns))
+
+(defn gen-datetime
+  "Generates datetime within given range and format.
+
+   base-datetime => By default it'll calculate the dates from year 2000.
+                    Generally this is a good idea instead of using (ct/now)
+                    since giving the same seed will generate the same output.
+                    If you would like to generate from a differnt base-datetime,
+                    Pass for example (ct/now) to use current time,
+                    Or pass a specific date-time (ct/date-time 2011 11 24)
+
+   offset-min & offset-max => The offset number range
+                              By default it is -1000 to 1000
+
+   offset-fns => List of functions which will be used with the given offset.
+                 It randomly picks one of the functions and
+                 applies the random offset with the given range.
+                 Check valid-offset-fns for possible values.
+                 By default its all the values of valid-offset-fns.
+
+   For example If you would like to generate datetime
+   from last 10 months to next 10 months:
+   (gen/sample (gen-datetime {:offset-fns [clj-time.core/months]
+                              :offset-min -10
+                              :offset-max 10}))
+   =>
+   (#<DateTime 1999-11-01T00:00:00.000Z>
+    #<DateTime 1999-12-01T00:00:00.000Z>
+    #<DateTime 2000-05-01T00:00:00.000Z>
+    ....)"
+  ([]
+   (gen-datetime {}))
+  ([{:keys [base-datetime offset-fns offset-min offset-max]
+     :or {offset-fns valid-offset-fns
+          offset-min -1000
+          offset-max 1000
+          base-datetime (ct/date-time 2000)}}]
```

Comment:
could pull this out into a `def` (could be private) so we're not constructing a new datetime on every call